hazelcast/src/main/java/com/hazelcast/query/impl/IndexImpl.java

```diff
@@ -93,10 +100,15 @@ public void saveEntryIndex(QueryableEntry e) throws QueryException {
             converter = attributeType == null ? TypeConverters.IDENTITY_CONVERTER : attributeType.getConverter();
         }
 
-        Data key = e.getIndexKey();
-        Comparable newValue = e.getAttribute(attribute);
+        Comparable oldValue = null;
+        if (oldRecordValue != null) {
+            oldValue = QueryEntry.extractAttribute(attribute, e.getKey(), oldRecordValue, ss);
```

Comment:
I don't want to change this part as it's subject of further changes because of the multi-value attribute.